mlflow/models/model.py

```diff
@@ -581,16 +585,18 @@ def to_dict(self):
             res.pop("run_id", None)
         return res
 
-    def to_yaml(self, stream=None):
+    def to_yaml(self, with_config: bool = True, stream=None):
```

Comment:
```suggestion
    def to_yaml(self, stream=None, with_config: bool = True):
```
to avoid breaking change.

Reply:
but I prefer to keep `to_dict`  / `to_yaml` intact, instead, we can drop certain fields in the result of `to_dict`. the `with_config` argument is ambiguous in the API.

Reply:
and I think we can only keep:

`flavors` that excludes `config`,
`utc_time_created`,
`artifact_path`

in model-history tag.

Reply:
So remove `with_config` from both `to_dict` and `to_yaml`? and for `mlflow.log-model.history` my understanding is that is that's done in `mlflow/store/tracking/sqlalchemy_store.py`. Instead of adding `model_dict = mlflow_model.to_dict()` as tag just keep `flavors` that excludes `config`, `utc_time_created` and `artifact_path` ?

Reply:
@WeichenXu123 when is release for `2.15.2`?